src/auditors/repository-rulesets.ts

```diff
@@ -1,21 +1,28 @@
-import { AuditorWarning, GraphqlRepository } from '../types';
-import { pluralize } from '../utils';
+import { AuditorFunction, AuditorWarning } from '../types';
 
 export const TYPE = 'repository-rulesets';
 
-export const auditor = async ({
-  graphqlRepository,
-}: {
-  graphqlRepository: GraphqlRepository;
+export const auditor: AuditorFunction = async ({
+  gitHubEnterpriseServerVersion,
+  octokit,
+  owner,
+  repo,
 }): Promise<AuditorWarning[]> => {
-  if (graphqlRepository.rulesets.totalCount > 0) {
```

Comment:
Using a shared GraphQL query isn't a good fit here - or at least will require tonnes of refactoring - because the `rulesets` node is only available on GHEC.